### PR TITLE
Combine various print result log events with different levels

### LINKED ISSUE #6173: [CT-1444] Structured logging: combine node_info events with different logging levels
Prior to this ticket a number of logging events had different event classes depending on the "level" of the logging. This makes it more difficult for consumers of the logs since they would have to subscribe to all versions of the events.

This ticket collapses the multiple events into single events.

This is spun off as part of the work being done in #5663 (CT-1048). Additional work will be done in that ticket to create proto messages for various nodes and node configs.

### PR DESCRIPTION
**Pull request by gshank**
resolves #6173

### Description

To simplify the work that consumers of our logs need to do, combine logging events issued at the same point in the code which differ mainly in the "level".

Part of the structured logging initiative. More to follow in #5663.

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

**Comment by gshank** [comment]
@jtcohen6 I added you as a reviewer to make sure that we're on the same page here, with collapsing the multiple events into one.

**Comment by jtcohen6** [comment]
@gshank I took a quick look through, and the consolidations proposed here make sense to me 👍 

If I understand right, events will no longer reflect their log levels by inheriting from level classes (`WarnLevel`, `ErrorLevel`), but now based on a new attribute of each event class containing the level. That makes sense insofar as proto schemas need to be composed, and do not support class inheritance. Do we have a clear sense of what's _lost_ by taking this approach? If I remember correctly, the class-first approach to structured logging a year ago was motivated by our ability to make use of `mypy` for type checking/enforcement — does the addition of proto schemas get us all that & better?

**Comment by gshank** [comment]
We do lose some of the mypy type checking, since the generated proto python classes all have defaults, so if a field is not supplied on the constructor, it doesn't cause an error. That would be something that would only be an issue during development, and if the generated logging event was not checked for validity. And there's really not much that can be done about it, since it's inherent in the way that protobuf works. 

This particular pull request just overrides the level for the changed logging events. The rest of the logging events remain the same, in that they are based on the various level classes.

**Review by emmyoop — COMMENTED** [review]
Is the plan to add `EventInfo` onto all proto events?  It's hard to tell why it is on some and not others.

Example: Why does `LogFreshnessResult` set the level when the message is generated in `freshness.py` but also have a level in the definition in `types.py`?  Or `LogHookStartLine` has `EventInfo` on the proto definition but `info` is not passed in.

**Comment by gshank** [comment]
It is true that if we *always* pass in the level with the "info" function, we wouldn't need to use a Level base class. I just left it there just in case, but if you think we should remove it...

**Comment by emmyoop** [comment]
> It is true that if we _always_ pass in the level with the "info" function, we wouldn't need to use a Level base class. I just left it there just in case, but if you think we should remove it...

I think that an event should either have a level or be dynamic.  And if it's dynamic it should not use a Level base class to make that clear.

**Comment by gshank** [comment]
I removed the Level base classes from the classes that specify the level.

File: .changes/unreleased/Under the Hood-20221028-110344.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+kind: Under the Hood
+body: Combine certain logging events with different levels
+time: 2022-10-28T11:03:44.887836-04:00
+custom:
+  Author: gshank
+  Issue: "6173"
+  PR: "6174"
```
[the remark this answers is not in the document]
Comment (gshank) [reply .changes/unreleased/Under the Hood-20221028-110344.yaml:7]: I think it depends on the ticket. This one actually changes the codes in the logs, so if anybody were checking them, it makes sense to flag it in the changelog

File: core/dbt/events/types.py [hunk 2]
```diff
@@ -1962,177 +1940,126 @@ def message(self) -> str:
 
 
 @dataclass
-class PrintModelResultLine(InfoLevel, pt.PrintModelResultLine):
+class LogModelResult(DynamicLevel, pt.LogModelResult):
```
Comment (emmyoop) [inline core/dbt/events/types.py:1943]: Love how clear this makes it now.

**Review by emmyoop — APPROVED** [review]

**Pull request merged**